src/vm.rs

```diff
@@ -683,18 +684,52 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
     }
 
     // Apply rewards (balance increments) to beneficiary accounts, etc.
-    fn apply_rewards(&self, write_set: &mut WriteSet, tx: &TxEnv, gas_used: U256) {
+    fn apply_rewards<DB: Database>(
+        &self,
+        write_set: &mut WriteSet,
+        tx: &TxEnv,
+        gas_used: U256,
+        #[allow(unused_variables)] evm_context: &EvmContext<DB>,
+    ) {
+        let mut gas_price = if let Some(priority_fee) = tx.gas_priority_fee {
+            std::cmp::min(tx.gas_price, priority_fee + self.block_env.basefee)
+        } else {
+            tx.gas_price
+        };
+        if self.spec_id.is_enabled_in(SpecId::LONDON) {
+            gas_price = gas_price.saturating_sub(self.block_env.basefee);
+        }
+
         let rewards: SmallVec<[(MemoryLocationHash, U256); 1]> = match self.reward_policy {
```

Comment:
We can make it `3` for OP here. Or even remove `SmallVec` and use array of 1 or 3 based on feature flag.

Reply:
I think we shouldn't.

In one block, there will only be _a single tx_ that uses 3 elements (to be more specific, the deposit transaction). The remaining N-1 blocks only uses _1_ element here.

Reply:
Okay, can keep this as is then. Can try `[MemoryLocationHash; 1]` for Ethereum later.